### DISCUSSION TITLE
Docker deployment with mainnet + regtest bundled as 1

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2676] [Ideas / Feature Requests]
I'm thinking about improving the Docker deployment so that is launches 2 BTCPayServer containers. One for mainnet + one for regtest, so shopkeepers can have a testing environment easily and have them run on the same version.

Later on, I'd try to write a bash script to copy data from production to regtest so the test environment can be kept in sync (and possibly anonymized).

Main things to consider are:
- Is this actually a good idea?
- Or would it be better to work on doing multi-docker deployments on the same host? I may be mistaken, but I don't think this is supported now, or is it?
- Create a docker fragment to enable this
- What containers can be used for both mainnest + regtest?
   - BTCPay Server itself obviously.
- What containers need to be created double
   - We need separate database containers (or put both in the same DB container?)
   - Double bitcoin core containers
   - Anything else?

**Comment by woutersamaey** [comment]
Can I get your thoughts on this @Kukks ? It would work alongside my efforts for easier testing.

**Comment by Zaxounette** [comment]
I have no problem with this but three points have to be taken into account:
* It must not be mandatory
* Even when activated, it must not overburden the hardware and significantly increase the cost of either the hardware, or the VPS hosting fees. Otherwise no one will use it, least of all in low income countries.
* Ideally, in the long run, should just be a 1 click activation/deactivation process in the maintenance tab (for example)

**Reply by woutersamaey** [reply]
100% agree. Same for the data copy process, have a button for that so that your test environment is easy to keep in sync.
